music21/musicxml/m21ToXml.py

```diff
@@ -3322,8 +3370,26 @@ def parseOneElement(self, obj):
         self.objectSpannerBundle = self.spannerBundle.getBySpannedElement(obj)
         preList, postList = self.prePostObjectSpanners(obj)
 
-        for sp in preList:  # directions that precede the element
-            root.append(sp)
+        if prePostObjectSpannersOnly:
+            if not preList and not postList:
+                return
+
+            for sp in preList:  # directions that precede the element
+                root.append(sp)
+
+            if postList and obj.quarterLength > 0:
+                # gap in stream before spanner stop: create <forward>
+                # Ignore Harmony objects having writeAsChord = False
+                if 'GeneralNote' in obj.classes and getattr(obj, 'writeAsChord', True):
```

Comment:
why GeneralNote and not Harmony?

Reply:
I was the last one to touch this; Greg is just duplicating it.

GeneralNote is the 99.9% case here. We're trying to _ignore_ any Harmony objects that have writeAsChord = False, hence the three-argument getattr() to ensure the attribute is either missing or True.

(I get that three-argument getattr can be a "buh?" though...)

Reply:
@jacobtylerwalls I'm still actually not sure I understand what you were trying to accomplish with the Harmony check in the "emit the note" case.  It looks like you skip updating self.offsetInMeasure for certain Harmony objects, but you still emit them?

Reply:
Yes, if writeAsChord is False, they need to be emitted (I don't know offhand which routine emits the proper tags for them), but they shouldn't take up any time (since they're not being written-as-chord, i.e. as `<note>` tags.)